src/Containers/Microsoft.NET.Build.Containers/Registry.cs

```diff
@@ -189,25 +188,22 @@ private async Task<ImageBuilder> ReadSingleImageAsync(string repositoryName, Man
         string reference,
         ManifestListV2 manifestList,
         string runtimeIdentifier,
-        string runtimeIdentifierGraphPath,
         CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
-        var runtimeGraph = GetRuntimeGraphForDotNet(runtimeIdentifierGraphPath);
-        var (ridDict, graphForManifestList) = ConstructRuntimeGraphForManifestList(manifestList, runtimeGraph);
-        var bestManifestRid = CheckIfRidExistsInGraph(graphForManifestList, ridDict.Keys, runtimeIdentifier);
-        if (bestManifestRid is null) {
-            throw new BaseImageNotFoundException(runtimeIdentifier, repositoryName, reference, graphForManifestList.Runtimes.Keys);
-        }
-        var matchingManifest = ridDict[bestManifestRid];
-        var manifestResponse = await GetManifestAsync(repositoryName, matchingManifest.digest, cancellationToken).ConfigureAwait(false);
+        IReadOnlyDictionary<string, PlatformSpecificManifest> ridDict = ConstructRuntimeGraphForManifestList(manifestList);
```

Comment:
Because we have a more regimented way to map RIDs, and because we as a platform are going away from a flexible RID graph, we can remove our use of the graph in favor of using the new mapper to 
a) identify RIDs supported by the platform-specific manifests, and
b) find the RID that matches best

Reply:
> Because we have a more regimented way to map RIDs, and because we as a platform are going away from a flexible RID graph, we can remove our use of the graph in favor of using the new mapper to
a) identify RIDs supported by the platform-specific manifests, and
b) find the RID that matches best

For non-portable rids used by source-build SDKs we still want the mapping to work from the non-portable rid to the portable rid (which then corresponds to an image platform os+arch).

That requires using the bundled runtime graph file.

Reply:
Good point, the conversation internally about the need for _some_ kind of graph did change over the past couple days. So a portion of this will likely come back, just with a different source graph.

Reply:
I was trying this out today, and the current code wasn't able to map a non-portable rid to a portable rid.
I haven't checked yet if the pebkac, but anyway could we use this function here:

https://github.com/dotnet/sdk/blob/741b910599db7f6a1a409fe70937af0f76caa437/src/Tasks/Microsoft.NET.Build.Tasks/NuGetUtils.NuGet.cs#L83-L84

Given a runtime graph, the target runtimeIdentifier, and the rids that correspond to the image platforms it will tell you what the best rid is.

Reply:
The only concern I have there is inflating the dependencies of this tooling - would need to check that packaging still works correctly. But I think the tests @vlada-shubina added around package layout would help us detect any problems.

Reply:
Resolving this conversation as I just merged this enhancement in another PR you've sent.